interfaces/abstract_task.h

```diff
@@ -32,12 +30,13 @@ class abstract_task {
   virtual void task(void); // this is the actual task
 
   string getInfo(void);
-  void set_SleepTime(int milliseconds);
 
 #if WithTaskSuspend == true
   TaskHandle_t getHandle() { return xHandle; };
 #endif
 
+  virtual void set_SleepTime(uint32_t milliseconds) = 0;
+  virtual uint32_t get_SleepTime() = 0;
   void sleep(void);
```

Comment:
These do not seem to be implemented. Maybe this is the actual issue with the deadlocks/startup getting stuck?

Reply:
In my opinion it is implemented in avstract:task.cpp, line 23

Reply:
No it is not. [abstract_task.cpp#L23](https://github.com/SolarEnergyRacers/solar_car_control_system/blob/feature/main-v4/interfaces/abstract_task.cpp#L23) implements the function `sleep()`. However, the functions `set_SleepTime` and `get_SleepTime` are neither implemented in `abstract_task`, nor by all classes that implement abstract_task. To give you an example where it is not implemented: https://github.com/SolarEnergyRacers/solar_car_control_system/blob/feature/main-v4/lib/IOExt2/IOExt2.cpp